How to use a private registry (#2027)
knative/docs: docs/serving/deploying/private-registry.md

---
title: "Deploying images from a private container registry"
linkTitle: "Deploying from private registries"
weight: 10
type: "docs"
---

Learn how to configure your Knative cluster to deploy images from a private 
container registry.

> To enable access to your private container images, you create a `imagePullSecret` type 

Reviewer: suggestion: "create a Kubernetes secret of type `imagePullSecret` with your credentials"

Or, if you leave it the way it is, "create a" --> "create an"

> To enable access to your private container images, you create a `imagePullSecret` type 

Reviewer: ```suggestion
To enable access to your private container images, create a `imagePullSecret` type 
```

Kubernetes secret with your credentials, add that secret to your default
[service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/),
and then deploy those configurations to your Knative cluster.

## Before you begin

> Ensure that you have the following setup and running:

Reviewer: ```suggestion
You need:
```

> - Knative Serving. [Learn about installing Knative](../../install/README.md)

Reviewer: ```suggestion
- A Kubernetes cluster with [Knative Serving installed](../../install/README.md)
```

> - You must have access to and the credentials to the private container registry 

Reviewer: ```suggestion
- The credentials to the private container registry 
```

  where your container image is stored.

## Configuring your credentials in Knative

> 1. Create a `imagePullSecret` type secret named `container-registry` that contains 

Reviewer: ```suggestion
1. Create a secret of type `imagePullSecret` named `container-registry` that contains 
```

   your credentials:

    ```shell
    kubectl create secret container-registry \
      --docker-server=[PRIVATE_REGISTRY_SERVER_URL] \
      --docker-email=[PRIVATE_REGISTRY_EMAIL] \
      --docker-username=[PRIVATE_REGISTRY_USER] \
      --docker-password=[PRIVATE_REGISTRY_PASSWORD]
    ```

    Where
    - `[PRIVATE_REGISTRY_SERVER_URL]` is the URL to the private
      registry where your container image is stored. 
       
       Examples:
       - Google Container Registry: [https://gcr.io/](https://gcr.io/)
       - DockerHub [https://index.docker.io/v1/](https://index.docker.io/v1/)
       
    * `[PRIVATE_REGISTRY_EMAIL]` is your email address that is associated with
      the private registry.
      
    * `[PRIVATE_REGISTRY_USER]` is your container registry username.

       Tip: 
       If you're using GCR and prefer to store and pull
       long-lived credentials instead of passing short-lived access tokens, see

>        [Authentication methods: JSON key file](/container-registry/docs/advanced-authentication#json_key_file).

Reviewer: This must be the link you mentioned needing to fix.

>        [Authentication methods: JSON key file](/container-registry/docs/advanced-authentication#json_key_file).

Reviewer 2: I would just remove this section. Every Docker Registry will likely have slightly different takes on username and passwords. I would rather let the user follow the advice of the registry provider.

Author: removed
